src/manifolds/Stiefel.jl

```diff
@@ -48,19 +48,52 @@ function PadeRetraction(m::Int)
     return PadeRetraction{m}()
 end
 @doc raw"""
-    CaleyRetraction <: AbstractRetractionMethod
+    CayleyRetraction <: AbstractRetractionMethod
 
 A retraction based on the Caley transform, which is realized by using the
 [`PadeRetraction`](@ref)`{1}`.
 """
-const CaleyRetraction = PadeRetraction{1}
+const CayleyRetraction = PadeRetraction{1}
 
-"""
-    CaleyVectorTransport <: AbstractVectorTransportMethod
+# This could maybe also be moved to Base.
+@doc raw"""
+    DifferentiatedRetractionVectorTransport{R<:AbstractRetractionMethod} <:
+        AbstractVectorTransportMethod
+
+A type to specify a vector transport that is given by differentiating a retraction.
+This can be introduced in two ways. Let ``\mathcal M`` be a Riemannian manifold,
+``p\in\mathcal M`` a point, and ``X,Y\in T_p\mathcal M`` denote two tangent vectors at ``p``.
+
+Given a retraction (cf. [`AbstractRetractionMethod`](@ref)) ``\operatorname{retr}``,
+the vector transport of `X` in direction `Y` (cf. [`vector_transport_direction`](@ref))
+by differentiation this retraction, is given by
+
+```math
+\mathcal T^{\operatorname{retr}}_{p,Y}X
+= D_Y\operatorname{retr}_x(Y)[X]
+= \frac{\mathrm{d}}{\mathrm{d}t}\operatorname{retr}_x(Y+tX)\Bigr|_{t=1}.
```

Comment:
```suggestion
= D_Y\operatorname{retr}_p(Y)[X]
= \frac{\mathrm{d}}{\mathrm{d}t}\operatorname{retr}_p(Y+tX)\Bigr|_{t=0}.
```
I think that's the correct formula?